packages/frontpage/middleware.ts

```diff
@@ -8,22 +8,33 @@ import {
 import { db } from "./lib/db";
 import * as schema from "./lib/schema";
 import {
-  deleteAuthCookie,
   getClientPrivateKey,
   getOauthClientOptions,
   getSession,
   importDpopJwks,
   signOut,
   oauthDiscoveryRequest,
+  getCookieJwt,
 } from "./lib/auth";
 
 export async function middleware(request: NextRequest) {
-  const session = await getSession();
-  if (!session) {
+  const cookieJwt = await getCookieJwt();
+  if (!cookieJwt) {
     return NextResponse.next();
   }
 
-  if (session.user.expiresAt.getTime() < new Date().getTime() - 500) {
+  // This check is for old cookies that don't have the exp field set
+  // Can be removed after a while (when all old cookies are expired)
+  if (!cookieJwt.payload.exp) {
+    await signOut();
```

Comment:
For some reason this throws an error of "Not authenticated" for existing sessions, somehow the session had already been deleted from the db.

Reply:
Attempting a fix that makes signOut idempotent.

Reply:
ccbb06eae3f8e40e1bc5473618512fb9dc82cd63